File: firework-admin-service/src/main/java/cn/fireworkstudio/admin/service/UserService.java
package cn.fireworkstudio.admin.service;

import cn.fireworkstudio.admin.dto.LoginReqDto;
import cn.fireworkstudio.admin.dto.LoginResDto;

/**
 * Copyright: 2016 FireworkStudio
 * *
 * Author:  Daniel Kong
 * Desc:    User Service
 */
public interface UserService {

    /**
     * User Login
     *
     * @param reqDto Login request
     * @return Login result
     */
    LoginResDto login(LoginReqDto reqDto);
}

File: firework-admin-web/src/main/webapp/js/services/common/user-service.js
'use strict';

/**
 * Copyright: 2016 FireworkStudio
 **
 * Author:  Daniel Kong
 * Desc:    user service
 */
adminApp.factory('userService', ['authService', 'restProxyService', 'API_PROVIDER', 'API_END_POINT', '$window',
    function (authService, restProxyService, API_PROVIDER, API_END_POINT, $window) {

        return {

            logIn: function (params) {
                return restProxyService.sendHttpPost(API_PROVIDER, API_END_POINT + '/user/login', params);
            },

            logOut: function () {
                authService.setUser(undefined);
                $window.location.href = '/admin/login.html';
            }
        }
    }]);